duplex_http_call.go

```diff
@@ -42,13 +42,11 @@ type duplexHTTPCall struct {
 	requestBodyReader *io.PipeReader
 	requestBodyWriter *io.PipeWriter
 
-	sendRequestOnce sync.Once
-	responseReady   chan struct{}
-	request         *http.Request
-	response        *http.Response
-
-	errMu sync.Mutex
-	err   error
+	requestSent   bool
```

Comment:
`sendRequestOnce` doesn't need to lock as `ensureMakeRequest` can't be called concurrently: https://github.com/connectrpc/connect-go/blob/b6d44ecfd51ddab5fa37fbbc7c3cbcece7940179/connect.go#L130-L134

Reply:
Reverted, changes were intended to reason about the racy behaviour but it's not a required fix.